cmd/gaia/app/sim_test.go

```diff
@@ -0,0 +1,129 @@
+package app
+
+import (
+	"encoding/json"
+	"math/rand"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	dbm "github.com/tendermint/tendermint/libs/db"
+	"github.com/tendermint/tendermint/libs/log"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banksim "github.com/cosmos/cosmos-sdk/x/bank/simulation"
+	"github.com/cosmos/cosmos-sdk/x/mock/simulation"
+	stake "github.com/cosmos/cosmos-sdk/x/stake"
+	stakesim "github.com/cosmos/cosmos-sdk/x/stake/simulation"
+)
+
+const (
+	defaultNumKeys   = 10
+	defaultNumBlocks = 100
+	defaultBlockSize = 100
+
+	simulationEnvEnable    = "GAIA_SIMULATION_ENABLED"
+	simulationEnvSeed      = "GAIA_SIMULATION_SEED"
+	simulationEnvKeys      = "GAIA_SIMULATION_KEYS"
+	simulationEnvBlocks    = "GAIA_SIMULATION_BLOCKS"
+	simulationEnvBlockSize = "GAIA_SIMULATION_BLOCK_SIZE"
```

Comment:
What was the thinking of using environment variables to pass information to the tests?

Reply:
I'm probably more into using test flags (https://stackoverflow.com/questions/47045445/idiomatic-way-to-pass-variables-to-test-cases-in-golang)

Reply:
OK, flags seem slightly cleaner - switched.